kofu/src/main/kotlin/org/springframework/fu/kofu/ConfigurationDsl.kt

```diff
@@ -58,7 +59,7 @@ open class ConfigurationDsl(private val dsl: ConfigurationDsl.() -> Unit): Abstr
 	 */
 	inline fun <reified T : Any> configurationProperties(properties: T? = null, prefix: String = ""): T {
 		val bindedProperties = properties ?: FunctionalConfigurationPropertiesBinder(context).bind(prefix, Bindable.of(T::class.java)).orElseGet { BeanUtils.instantiateClass(T::class.java) }
-		context.registerBean<T>("${T::class.java.simpleName.toLowerCase()}ConfigurationProperties") {
+		context.registerBean<T>("${T::class.java.simpleName.lowercase(Locale.getDefault())}ConfigurationProperties") {
```

Comment:
For what default locale is needed here?
Shouldn't it be simple lowercase()?

Reply:
toLowerCase is now deprecated in Kotlin 1.5.0, its Dokka doc describe it as `Returns a copy of this string converted to lower case using the rules of the default locale.`, the deprecation replacement tip suggest to change to :
```
ReplaceWith("lowercase(Locale.getDefault())", "java.util.Locale")
```
That's why I didi it :)

lowercase() doc is a little different : `Returns a copy of this string converted to lower case using Unicode mapping rules of the invariant locale.`

Reply:
I think lowercase() is exactly what needed in this case - locale insensitive transformation.

Reply:
I made the change